contributing.md

```diff
@@ -0,0 +1,163 @@
+* Think you have a nice config or unique use-case for beets? We'd love to hear about it! Submit a post to our [our forums](https://discourse.beets.io/) for a chance to get featured in our blog or use-case page.
+* Consider helping out in [our forums](https://discourse.beets.io/) by responding to support requests or driving some new discussions. 
+
+[docs]: http://beets.readthedocs.org/
+
+## Programming contributions
+* As a programmer (even if you're just a beginner!), you have a ton of opportunities to get your feet wet with `beets`. 
+* For developing plugins, or hacking away at beets, there's some good information in the [for developers](https://beets.readthedocs.io/en/stable/dev/) section of the docs.
+
+### Getting the Source
+
+The easiest way to get started with the latest beets source is to use [pip](http://pip.openplans.org/)  to install an "editable" package. This can be done with one command:
+
+    $ pip install -e git+https://github.com/beetbox/beets.git#egg=beets
+
+Or, equivalently:
+
+    $ git clone https://github.com/beetbox/beets.git
+    $ cd beets
+    $ pip install -e .
+
+If you already have a released version of beets installed, you may need to
+remove it first by typing `pip uninstall beets`. The pip command above will put
+the beets source in a `src/beets` directory and install the `beet` CLI script to
+a standard location on your system. You may want to use the `--src` option to specify
+the parent directory where the source will be checked out and the `--user` option
+such that the package will be installed to your home directory (compare with the output of
+`pip install --help`). 
+
+Alternatively, you can get the source via 
+[git at GitHub](http://github.com/beetbox/beets). The pip method above installs
+dependencies automatically, but if you do it manually, you can use pip to
+install these Python modules:  
+* `mutagen`
+* `munkres`
+* `unidecode`
+* `pyYAML`
+* `musicbrainzngs`
+* `confuse`
+* `mediafile`
+* `jellyfish`
+
+Please be aware that this list might be out of date, but [setup.py](https://github.com/beetbox/beets/blob/60bba370c0ecc58b4057d78867273e5936d22dee/setup.py#L89-L110) contains a list that is always up to date. With all the required modules in place, you can just type `./beet` in the source directory to run beets from there.
+
+
+### Code contribution ideas
+* We maintain a set of [issues marked as "bite-sized"](https://github.com/beetbox/beets/labels/bitesize). These are issues that would serve as a good introduction to the codebase. Claim one and start exploring!
+* Our [test coverage](https://codecov.io/github/beetbox/beets) is somewhat low. You can help out by finding low-coverage modules on Coveralls and adding new tests.
+* There are several ways to improve the tests in general (see [Testing](https://github.com/beetbox/beets/wiki/Testing)) and some places to think about performance optimization (see [Optimization](https://github.com/beetbox/beets/wiki/Optimization)).
```

Comment:
I noticed this mentions use of Nose, which is now unmaintained. Is this current info, or should beets perhaps consider other avenues? I could have sworn I've seen talk on this before, but I can't seem to find it now.

Reply:
I don't have a link to a previous thread, but the way things stand are that I actually do tend to use nose even though it's unmaintained. I should probably switch to pytest along with the rest of the world. But I do kind of like the status quo where our tests are framework-agnostic, in the sense that nose is *just* a test runner and pytest (or `python -m unittest` or [ptr](https://github.com/facebookincubator/ptr) or whatever) works just as well.

Reply:
Yeah that's true, I'll see if I can make some changes to the Testing wiki to clarify some of this stuff.